### PR TITLE
feat: invalidate all sessions of a user after password changed

### LINKED ISSUE #5753: 期望在用户密码修改之后能够移除活跃的登录 Session
### 你当前使用的版本

2.14.0

### 描述一下此特性

目前用户修改密码之后不会移除当前已登录的 Session，这可能会导致一定的安全问题，因为修改密码之后，在另外的活跃登录页面中仍然能够再次修改密码或者其他资料。所以期望能够在修改密码之后移除掉活跃的登录 Session。

/area core
/kind feature
/milestone 2.15.x

### 附加信息

_No response_

### PR DESCRIPTION
**Pull request by guqing**
#### What type of PR is this?
/kind feature
/area core
/milestone 2.15.x

#### What this PR does / why we need it:
新增修改用户密码之后踢除用户的所有登陆会话的功能

#### Which issue(s) this PR fixes:

Fixes #5753

#### Does this PR introduce a user-facing change?

```release-note
新增修改用户密码之后踢除用户的所有登陆会话的功能
```

### CODE REVIEW

File: application/src/main/java/run/halo/app/security/session/ReactiveFindByIndexNameSessionRepository.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.session.Session;
+import reactor.core.publisher.Mono;
+
+/**
+ * TODO: Wait for <code>spring session 3.3 released</code> and then remove this interface.
+ * if you want to remove this interface, please modify the {@link PrincipalNameIndexResolver} in
+ * {@link InMemoryReactiveIndexedSessionRepository} to have a parameterized constructor(since
+ * spring session 3.3) and overload the findByPrincipalName method.
+ */
+public interface ReactiveFindByIndexNameSessionRepository<S extends Session> {
```
Comment (JohnNiang) [inline application/src/main/java/run/halo/app/security/session/ReactiveFindByIndexNameSessionRepository.java:15]: Hi @guqing ，I have proposed another PR to [upgrade Spring Boot to 3.3.0-RC1](https://github.com/halo-dev/halo/pull/5760) which includes Spring Session 3.3.

The interface can be removed after merging https://github.com/halo-dev/halo/pull/5760.

Comment (guqing) [reply application/src/main/java/run/halo/app/security/session/ReactiveFindByIndexNameSessionRepository.java:15]: ok

**Review by LIlGG — COMMENTED** [review]
/lgtm

**Review by ruibaby — APPROVED** [review]
/lgtm

**Pull request merged**